core/amazon/index.js

```diff
@@ -150,7 +153,13 @@ var amazonProduct = {
                                 var variation = variationKeys[variationIdx];
                                 var value = itemAttributes[variation];
                                 if (!(value in ref)) {
+                                    // Otherwise, add value to map
                                     if (variationIdx == variationKeys.length - 1) {
+                                        // Check if the last level variations are too numerous for selection through conversation
+                                        if (Object.keys(ref).length == 10) {
+                                            conversational = false;
```

Comment:
I think it should be if(Object.keys(ref).length > 10) to mark conversational to be false. And also move the if statement after pushing the new item in ref.